### PR TITLE
 WPF - Fix the Chinese IMEs (such as "sogou pinyin", "qq...") window position is incorrect in Winform form embedded wpf control.

### PR DESCRIPTION
**Pull request by feliciwithme**
**Summary:** 
 - Fix the WPF browser is embedded in the Winfrom form, the Chinese IMEs window (such as "sogou pinying","qq...") will not follow the cursor when inputting Chinese. This problem does not exist in the WPF form.
 
**Changes:**
 - Modify the WpfImeKeyboardHandler class and add a custom function GetOutermostElement to obtain the outermost visual element.
 - Modify the ChangeCompositionRange method and add that when the parent window Window.GetWindow(owner) is empty, call the function GetOutermostElement to obtain the outermost visual element as the input method window position calculation container.
      
**How Has This Been Tested?**  

**Screenshots (if appropriate):**
**The wrong position**
image
**The correct position**
After the fix, the input method window correctly follows the cursor position.
image

**Types of changes**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Updated documentation

**Checklist:**

- [x] Tested the code(if applicable)
- [x] Commented my code
- [ ] Changed the documentation(if applicable)
- [ ] New files have a license disclaimer
- [ ] The formatting is consistent with the project (project supports .editorconfig)

### CODE REVIEW

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 using Point = System.Windows.Point;
 using Range = CefSharp.Structs.Range;
 using Rect = CefSharp.Structs.Rect;
-
+using System.Windows.Media;
```
Comment (amaitland) [inline CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs:16]: Fix up spacing  here. There should be a line between usings and namespace

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs [hunk 2]
```diff
@@ -40,6 +40,25 @@ public WpfImeKeyboardHandler(ChromiumWebBrowser owner) : base(owner)
         {
         }
 
+        /// <summary>
+        /// Get the outermost element of the browser 
+        /// </summary>
+        /// <param name="control">The browser</param>
+        /// <returns>The outermost element </returns>
+        FrameworkElement GetOutermostElement(FrameworkElement control)
```
Comment (amaitland) [inline CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs:48]: Please make this method private and move it to the bottom of the class with the other private methods.

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs [hunk 3]
```diff
@@ -61,7 +80,8 @@ public void ChangeCompositionRange(Range selectionRange, Rect[] characterBounds)
             {
                 //TODO: Getting the root window for every composition range change seems expensive,
                 //we should cache the position and update it on window move.
-                var parentWindow = Window.GetWindow(owner);
+                //In Winform embedded wpf borwser mode, Window.GetWindow(owner) is null, so use a custom function to get the outermost visual element.
+                var parentWindow = GetOutermostElement(owner);
```
Comment (amaitland) [inline CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs:84]: Please keep the existing behaviour.

If the `parentWindow` is null then fallback to calling `GetOutermostElement`

**Review by amaitland — CHANGES_REQUESTED** [review]
Thanks. Comments inline

**Comment by amaitland** [comment]
Please edit the description and complete the required sections

> **Summary:** [summary of the change and which issue is fixed here]

Complete this section

> **Changes:** [specify the structures changed]

Complete this section

> How Has This Been Tested?

Complete this section

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs (hunk 2 above)
Comment (feliciwithme) [reply CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs]: ok.

**Comment by amaitland** [comment]
Few minor changes still, please also edit the description as per above.

**Comment by feliciwithme** [comment]
> Few minor changes still, please also edit the description as per above.

Thank you, I have resubmitted the running comparison results.

**Comment by amaitland** [comment]
Please fix the styling issues. Once that's done this can be merged.

**Comment by feliciwithme** [comment]
> 请解决样式问题。一旦完成，就可以合并。

I have modified the style problem, but I am not sure if it meets the requirements? If not, please point it out, thank you.

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs [hunk 4]
```diff
@@ -40,6 +40,25 @@ public WpfImeKeyboardHandler(ChromiumWebBrowser owner) : base(owner)
         {
         }
 
+        /// <summary>
+        /// Get the outermost element of the browser 
+        /// </summary>
+        /// <param name="control">The browser</param>
+        /// <returns>The outermost element </returns>
+        private FrameworkElement GetOutermostElement(FrameworkElement control)
```
Comment (amaitland) [inline CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs:48]: Please move private method to bottom of class.

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs [hunk 5]
```diff
@@ -10,10 +10,10 @@
 using CefSharp.Internals;
 using CefSharp.Structs;
 using CefSharp.Wpf.Internals;
+using System.Windows.Media;
 using Point = System.Windows.Point;
 using Range = CefSharp.Structs.Range;
 using Rect = CefSharp.Structs.Rect;
-
```
Comment (amaitland) [inline CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs:16]: The line between the namespace and usings has been removed. Please add a new line between namespace and usings

**Review by amaitland — CHANGES_REQUESTED** [review]

File: CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs (hunk 4 above)
Comment (feliciwithme) [reply CefSharp.Wpf/Experimental/WpfIMEKeyboardHandler.cs]: Got it, modified.

**Pull request merged**